src/composables/transferSendForm.ts

```diff
@@ -0,0 +1,158 @@
+import {
+  computed,
+  ref,
+} from 'vue';
+import { useI18n } from 'vue-i18n';
+import { useForm } from 'vee-validate';
+
+import type {
+  Dictionary,
+  IAsset,
+  IToken,
+  TransferFormModel,
+} from '@/types';
+import { useModals } from '@/composables/modals';
+import {
+  APP_LINK_WEB,
+  IS_PRODUCTION,
+  MODAL_READ_QR_CODE,
+} from '@/constants';
+import { toShiftedBigNumber, getMessageByFieldName } from '@/utils';
+
+type SelectedAssetValueFunction = (token?: string, selectedAsset?: IAsset) => IToken | undefined;
+
+interface UseTransferSendFormParams {
+  transferData: TransferFormModel;
+  getSelectedAssetValue?: SelectedAssetValueFunction;
+}
+
+export function useTransferSendForm({
+  transferData,
+  getSelectedAssetValue,
+}: UseTransferSendFormParams) {
+  const formModel = ref(transferData);
+  const invoiceId = ref(null);
+  const invoiceContract = ref(null);
+
+  const { t } = useI18n();
+  const { openModal, openDefaultModal } = useModals();
+  const { errors, validate } = useForm();
+
+  const hasError = computed((): boolean => ['address', 'amount'].some((errorKey) => getMessageByFieldName(errors.value[errorKey]) === 'error'));
+
+  function clearPayload() {
+    formModel.value.payload = '';
+  }
+
+  function handleAssetChange(selectedAsset: IToken) {
+    formModel.value.selectedAsset = selectedAsset;
+  }
+
+  function returnFormModelValues({
+    account,
+    amount,
+    payload,
+    token,
+  }: Dictionary, tokenHandler?: SelectedAssetValueFunction) {
```

Comment:
If you split the arguments into separate lines I suggest to do it with all the arguments.
```suggestion
  function returnFormModelValues(
    { account, amount, payload, token }: Dictionary,
    tokenHandler?: SelectedAssetValueFunction,
) {
```